### ISSUE TITLE
Blank characters in configuration names not handled properly in gmake and gmake2

### ISSUE THREAD

**Issue by utelle** [premake/premake-core#1321]
If configuration names contain blank characters, like for example `"DLL Debug"` or `"DLL Release"`, then the actions `gmake` and `gmake2` do not generate proper path names for `OBJDIR` and `TARGETDIR`.

`gmake` escapes blank characters with `\ `, but this doesn't help. On running `make` directories are not created properly, because the directory name is truncated after the first blank, resulting in various error messages.

`gmake2` does not perform any escaping at all. So the effect is the same as for `gmake`: directories are not created properly.

As a workaround I modified my local premake5 installation so that blanks are replaced by underscores. I'm not sure whether this is a reasonable approach. Nevertheless, the actions `gmake` and `gmake2` should be changed to generate makefiles that work even if pathnames for `OBJDIR` or `TARGETDIR` contain blank characters, as is the case if configuration names contain blanks.

**Comment by matheusalbino** [comment]
I made tests and this case really happens, searching in the source code and found the possible origin of this bug.

When you define **objdir** or **targetdir** with double backslashes(``\\``) for fix the problem on premake5.lua, the functions ``path.getabsolute`` and ``path.getrelative`` remove the double backslashes, causing the bug.

I searching more forward, i found the functions causing this problem:
- ``do_translate``  in ``do_getabsolute`` used by ``path_getabsolute``
- ``path_normalize``  in ``do_normalize`` used by ``path_getrelative``

When the path comes to the C functions, her don't have more double backslashes, have only one backslash, and when passed to normalize or translate function the backslash is removed, prevented to inserted on Makefile for resolve the directory correctly.

**Comment by starkos** [comment]
> When you define objdir or targetdir with double backslashes(\\) for fix the problem on premake5.lua

Yeah, that's not going to fix anything. The problem is that the paths aren't properly escaped/quoted at generation-time, and there isn't anything you going to be able to do on the scripting side to fix it. This will require a PR to add the quotes/escaping to the makefile generator.

**Comment by utelle** [comment]
Unfortunately I don't know much about makefile generation. That's why I use premake in the first place. Therefore I fear I will not be able to provide a PR for fixing this problem.

Looking at the `make` documentation I get the impression that it is not trivial to properly mask space characters. It seems to be recommended to avoid space characters in file and directory names. So maybe the best approach is maybe to replace space characters by for example underscores in the definition of OBJDIR and TARGETDIR.
